tiledb/sm/query/readers/dense_reader.cc

```diff
@@ -583,54 +635,155 @@ void DenseReader::init_read_state() {
   read_state_.initialized_ = true;
 }
 
+/**
+ * Compute the maximum tile coords that we can process on this itereation to
+ * respect the memory budget.
+ */
+template <class DimType>
+tuple<uint64_t, std::vector<ResultTile*>> DenseReader::compute_result_tiles(
+    const std::vector<std::string>& names,
+    const std::unordered_set<std::string>& condition_names,
+    Subarray& subarray,
+    uint64_t t_start,
+    std::map<const DimType*, ResultSpaceTile<DimType>>& result_space_tiles) {
+  // For easy reference.
+  const uint64_t left_over_budget = std::min(
+      tile_memory_budget_,
+      memory_budget_ - array_memory_tracker_->get_memory_usage());
+  const auto& tile_coords = subarray.tile_coords();
+
+  // Keep track of the required memory to load the result space tiles. The first
+  // element of this vector is an aggregate of all query condition fields,
```

Comment:
Style nit: Is there any reason not to put this in a separate `total_condition_memory` variable? I really don't like vectors with implicit slot meaning [1]. I know it's only local to this function, but it still makes the code more difficult to understand.

[1] Same complaint as the buffer handling elsewhere, where the slot interpretation is determined by whether a field is var or nullable.

Reply:
We can't have only one variable. As we process attributes one by one, only the memory per attribute is relevant to respect the memory budget (Here attribute in the query condition get loaded all at once so that's the exception). With var sized tiles, we cannot know which attribute will be bigger than another and an attribute that was bigger than another after a few tiles might not stay the biggest attribute as we add more. So we have to keep track by attributes. I could have used a map to not have implicit slot meaning but I'm worried that the constant string lookups could cause performance issues. We could make classes that make the slot meaning consistent across multiple usages. Thoughts?

Reply:
Yes, the vector per-attribute is fine -- I meant specifically the `required_memory[0]` slot mentioned here for aggregation. If we can replace it with `total_condition_memory`, we would also avoid the `+1` offsetting below. Clearer IMO.

Reply:
Gotcha! Sorry I misread the initial comment... Fixed.

Reply:
Thanks! LGTM.